compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/IREEGPUAttrs.h

```diff
@@ -20,27 +20,164 @@
+//    Just like the element-tiles, the layout of these "outer tiles" is
+//    row-major in the sense that the last-enumerated semantic dimension is the
+//    most-contiguous dimension. The outer dimensions describe the layout of
+//    element tiles in the overall vector operand of the intrinsic. This is used
+//    only by a minority of intrinsics to describe "non-contiguous" operand
+//    tiles.
+//
+// Example: MFMA_I32_32x32x8_I8 accumulator
+// ----------------------------------------
+//
+// For the accumulator operand of MFMA_I32_32x32x8_I8, we have:
+//
+//   outer    = {4,  1}
+//   thread   = {2,  32}
+//   tstrides = {32, 1}
+//   element  = {4,  1}
+//
+// Let us first check the general invariants:
+// 0. The arrays all have length 2, corresponding to the semantic rank 2 of
+//    matmul accumulators, where the 2 semantic dimensions are M and N.
+// 1. The semantic tile size is (M = 32, N = 32) and that does match the product
+//    of the corresponding array elements outer[i] * thread[i] * element[i],
+//    for instance 4 * 2 * 4 == 32.
+// 2. The product of all the outer[i] times all the element[i] is:
+//    4 * 1 * 4 * 1 == 16.
+//    This corresponds to the @llvm.amdgcn.mfma.i32.32x32x8i8 intrinsic's vector
+//    operand type, <16 x i32>.
+// 3. The product of the thread[i] is 2 * 32 == 64. This corresponds to the
+//    subgroup size 64 on CDNA3. It could also have been a divisor of that, but
+//    here the exact match means that each thread receives a different tile.
+//
+// The tstrides here are such that the distribution of element-tiles to threads
+// is "row-major": the fact that tstrides[1] == 1 means that as we move by
+// one element tile down the N-dimension, we move to the next thread by tid.
+//
+// Let us now detail the exact layout:
+// 1. Always start by looking at element[]. Here we see that the element tile
+//    has shape 4x1.
+// 2. The thread-grid has shape 2x32, and as noted above has row-major thread
+//    distribution. As what is being distributed to threads here is element
+//    tiles of shape 4x1, at this point we have distributed an overall 8x32
+//    tile.
+// 3. Finally, the outer[] completes the picture by saying that those 8x32
+//    tiles are stacked vertically to form the overall 32x32 tile. The fact that
+//    the vector length 16 is split into outer[0]==4 and element[0]==4 means
+//    that these vectors contain groups of 4 matrix elements that are contiguous
+//    along the M-dimension (the "element tiles"), but that there is a
+//    discontinuity as the next 4 elements (the next "element tile") comes from
+//    far away elsewhere in the C matrix, owing to the fact that thread[0] is
+//    greater than 1. Example: thread 0 gets this accumulator ("C") operand:
+//    { C[0, 0], C[1, 0], C[2, 0], C[3, 0],
+//      C[8, 0], C[9, 0], C[10, 0], C[11, 0],
+//      C[16, 0], C[17, 0], C[18, 0], C[19, 0],
+//      C[24, 0], C[25, 0], C[26, 0], C[27, 0] }
```

Comment:
nit: Align the elements of the matrix?
```suggestion
//    { C[0, 0],  C[1, 0],  C[2, 0],  C[3, 0],
//      C[8, 0],  C[9, 0],  C[10, 0], C[11, 0],
//      C[16, 0], C[17, 0], C[18, 0], C[19, 0],
//      C[24, 0], C[25, 0], C[26, 0], C[27, 0] }
```